File: Data Structures & Algorithms/permutation-string/submission-0.py
class Solution:
    def checkInclusion(self, s1: str, s2: str) -> bool:
        d = {}
        for i in s1:
            if i in d: d[i]+=1
            else: d[i]=1
        
        d2 = {}
        for i in range(len(s2)):
            if s2[i] in d2: d2[s2[i]]+=1
            else: d2[s2[i]]=1

            if i>=len(s1)-1:
                ans = True
                if i>=len(s1): d2[s2[i-len(s1)]]-=1
                for k in d2:
                    if k not in d:
                        if d2[k]!=0:
                            ans = False
                            break
                    elif d[k]!=d2[k]:
                        ans = False
                        break
                if ans: return True
            print(d,d2)
        return False

File: Data Structures & Algorithms/k-closest-points-to-origin/submission-0.cpp
class Solution {
public:
    vector<vector<int>> kClosest(vector<vector<int>>& points, int k) {
        priority_queue<pair<int, pair<int,int>>,
                       vector<pair<int, pair<int,int>>>,
                       greater<pair<int, pair<int,int>>>> pq;
        for(auto i : points){
            pq.push({i[0]*i[0] + i[1]*i[1], {i[0],i[1]}});
        }
        vector<vector<int>> ans(k, vector<int>(2));
        for(int i = 0 ; i < k ; i++){
            ans[i][0] = pq.top().second.first;
            ans[i][1] = pq.top().second.second;
            pq.pop();
        }
        return ans;
    }
};
